docs/PAYLOAD.md

```diff
@@ -72,6 +72,115 @@ Some ways to handle this *double* event are:
 - send two pushes, one to be processed in the background, and the other to show up in the shade.
 - include a unique ID in your push so you can check to see if you've already processed this event.
 
+# Message Format Overview
+
+## Android Message Format
+
+The JSON push message can contain the following fields, see https://developers.google.com/cloud-messaging/http-server-ref for a complete list.
+
+```javascript
+var content = {
+	"priority": "normal", // Valid values are "normal" and "high."
+	"content_available": "0", // see "Background Notifications" below
+	"data": {
+		"title": "A short string describing the purpose of the notification",
+		"message": "The text of the alert message", // "body" can be used as alias, is converted to "message"
+		// localozation of message is possible
+		"count": 5, // set the badge notification count at app icon
+		"notId": 1, // is it necessary to set this?
+		"custom_key1": "value1",
+		"custom_key2": "value2"
+	},
+	// "notification": { ... } // can be used instead of data for title and message, see ....
+}
+```
+
+This is the JSON-encoded format you can e.g. send via AWS-SNS's web UI.
+Note, that the core message is json-encoded twice, so if we take the `content` from above you convert it this way
+
+```javascript
+var gcm_message = JSON.stringify({
+	"GCM": JSON.stringify(content),
+	"default": "plain text message again"
+	};
+```
+
+```json
+{"GCM":"{\"priority\":\"normal\",\"content_available\":\"0\",\"data\":{\"title\":\"A short string describing the purpose of the notification\",\"message\":\"The text of the alert message\",\"count\":5,\"notId\":1,\"custom_key1\":\"value1\",\"custom_key2\":\"value2\"}}"}
+```
+
+This message is received in the "notification" handler as follows.
+Note that the properties are "normalized" accross platforms, so this is passed to the app on android:
+
+```json
+{
+	"count": "5",
+	"title": "A short string describing the purpose of the notification",
+	"message": "The text of the alert message",
+	"additionalData": {
+		"custom_key1": "value1",
+		"custom_key2": "value2",
+		"notId": "1",
+		"dismissed": false,
+		"google.message_id": "...",
+		"coldstart": false,
+		"foreground": false
+	}
+}
+```
+
+## iOS Message Format
+
+The JSON message can contain the following fields, see [Apple developer docs](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/PayloadKeyReference.html#//apple_ref/doc/uid/TP40008194-CH17-SW5) for a complete list
+
+```javascript
+{
+    "aps":{
+        "alert": { // alternatively just a string: "Your Message",
+			"title": "A short string describing the purpose of the notification",
```

Comment:
The blocks above are properly formatted, but the ones bellow are not. I think we could use a proper json formatting through all examples/snippets.

Reply:
You mean this one:

`{"GCM":"{\"priority\":\"normal\",\"content_available\":\"0\",\"data\":{\"title\":\"A short string describing the purpose of the notification\",\"message\":\"The text of the alert message\",\"count\":5,\"notId\":1,\"custom_key1\":\"value1\",\"custom_key2\":\"value2\"}}"}`?

The problem here is that this is a very long string and AFAI it cannot be split into several lines in JSON. I could write

`{
"GCM":
"{\"priority\":\"normal\",\"content_available\":\"0\",\"data\":{\"title\":\"A short string describing the purpose of the notification\",\"message\":\"The text of the alert message\",\"count\":5,\"notId\":1,\"custom_key1\":\"value1\",\"custom_key2\":\"value2\"}}"
}`

but I fear this doesn't help much ... what do you think?

Reply:
I was actually refering to the "normal" payload sections. There were a few parts where an inner object was indented with tabs instead of spaces, and even though I don't have a problem with tabs per se, they have mixed whitespace usage. We should stick to one of them for consistency.

The "generated" payload examples you provided need not be formatted, I agree on that.

Reply:
alright, I'll fix the whitespaces, didn't notice.

Reply:
I just checked in a version with corrected whitespace and "sound" attribute added. What do you think now?